core/dbt/include/global_project/macros/materializations/models/incremental/merge.sql

```diff
@@ -86,14 +86,16 @@
 {%- endmacro %}
 
 
-{% macro get_insert_overwrite_merge_sql(target, source, dest_columns, predicates, include_sql_header=false) -%}
-  {{ adapter.dispatch('get_insert_overwrite_merge_sql', 'dbt')(target, source, dest_columns, predicates, include_sql_header) }}
+{% macro get_insert_overwrite_merge_sql(target, source, dest_columns, predicates) -%}
+  {{ adapter.dispatch('get_insert_overwrite_merge_sql', 'dbt')(target, source, dest_columns, predicates) }}
 {%- endmacro %}
 
-{% macro default__get_insert_overwrite_merge_sql(target, source, dest_columns, predicates, include_sql_header) -%}
+{% macro default__get_insert_overwrite_merge_sql(target, source, dest_columns, predicates) -%}
     {%- set predicates = [] if predicates is none else [] + predicates -%}
     {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
     {%- set sql_header = config.get('sql_header', none) -%}
+    {# The following is only true in BigQuery #}
```

Comment:
I like the way you did this!

Maybe even more explicit in the code annotation:
```
    {#-- The only time include_sql_header = True: --#}
    {#-- BigQuery + insert_overwrite strategy + "static" partitions config --#}
```

Reply:
Done!